csrc/kernel_cache.cpp

```diff
@@ -437,9 +437,11 @@ void prepareRuntimeOrder(
 
 FusionExecutorCache::FusionExecutorCache(
     std::unique_ptr<Fusion> fusion,
+    Communicator* communicator,
```

Comment:
It feels a bit cumbersome to pass `communicator` around like this. Would you prefer singleton instead? That would make code simpler and automatically fix deserializing a multi-GPU FusionExecutorCache, which this PR has been ignoring so far. I'm not aware of a plan to use multiple communicators in a process, but I could well be missing something.

Reply:
singleton sounds great! As you say, having multiple `Communicator`s has no benefit (for now).

For the record: https://github.com/NVIDIA/Fuser/pull/2243#discussion_r1601937356

Reply:
FYI, unfortunately, 1064e7622bb6f66fe5aa3563c794c1e9941e622a hit some practical problems related to destruction order. I could work around this with explicitly deleting a singleton, but it would make it hard to reason about the lifetime of the Communicator and doesn't seem worthwhile. I'll probably stick with passing around pointers for now.